### PR TITLE
Switched DeBruijnEnv to BRAL.indexone impl.

### PR DESCRIPTION
**Pull request by bezirg**
Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

### CODE REVIEW

**Comment by michaelpj** [comment]
Seems good!

**Comment by kwxm** [comment]
This is great! I have a branch that I opened on the 30th of November last year where I've been checking the consistency of the benchmark results (they're very consistent when run from here).  Here's how the times from back then compare with the times in this PR.

```
Script                         30th Nov       9 Feb       Change
----------------------------------------------------------------------------------
auction_1-1                    403.4 μs      298.0 μs     -26.1%
auction_1-2		       1.190 ms	     965.9 μs     -18.8%
auction_1-3		       1.199 ms	     963.1 μs     -19.7%
auction_1-4		       539.8 μs	     392.7 μs     -27.3%
auction_2-1		       407.2 μs	     299.4 μs     -26.5%
auction_2-2		       1.191 ms	     967.2 μs     -18.8%
auction_2-3		       1.496 ms	     1.226 ms     -18.0%
auction_2-4		       1.190 ms	     962.2 μs     -19.1%
auction_2-5		       536.4 μs	     391.7 μs     -27.0%
crowdfunding-success-1	       480.8 μs	     352.6 μs     -26.7%
crowdfunding-success-2	       480.6 μs	     352.4 μs     -26.7%
crowdfunding-success-3	       480.7 μs	     352.3 μs     -26.7%
currency-1		       488.2 μs	     380.5 μs     -22.1%
escrow-redeem_1-1	       746.3 μs	     558.3 μs     -25.2%
escrow-redeem_1-2	       746.9 μs	     557.4 μs     -25.4%
escrow-redeem_2-1	       854.0 μs	     654.7 μs     -23.3%
escrow-redeem_2-2	       852.4 μs	     653.8 μs     -23.3%
escrow-redeem_2-3	       852.1 μs	     653.4 μs     -23.3%
escrow-refund-1		       356.2 μs	     260.3 μs     -26.9%
future-increase-margin-1       490.4 μs	     380.7 μs     -22.4%
future-increase-margin-2       1.082 ms	     845.3 μs     -21.9%
future-increase-margin-3       1.081 ms	     845.5 μs     -21.8%
future-increase-margin-4       978.0 μs	     774.4 μs     -20.8%
future-increase-margin-5       1.432 ms	     1.172 ms     -18.2%
future-pay-out-1	       488.4 μs	     381.9 μs     -21.8%
future-pay-out-2	       1.079 ms	     845.7 μs     -21.6%
future-pay-out-3	       1.078 ms	     844.8 μs     -21.6%
future-pay-out-4	       1.424 ms	     1.167 ms     -18.0%
future-settle-early-1	       488.3 μs	     380.0 μs     -22.2%
future-settle-early-2	       1.078 ms	     845.0 μs     -21.6%
future-settle-early-3	       1.079 ms	     847.0 μs     -21.5%
future-settle-early-4	       1.129 ms	     907.2 μs     -19.6%
game-sm-success_1-1	       820.4 μs	     625.7 μs     -23.7%
game-sm-success_1-2	       457.0 μs	     331.6 μs     -27.4%
game-sm-success_1-3	       1.206 ms	     964.9 μs     -20.0%
game-sm-success_1-4	       534.3 μs	     388.9 μs     -27.2%
game-sm-success_2-1	       825.3 μs	     624.9 μs     -24.3%
game-sm-success_2-2	       457.8 μs	     332.7 μs     -27.3%
game-sm-success_2-3	       1.208 ms	     970.5 μs     -19.7%
game-sm-success_2-4	       534.5 μs	     388.3 μs     -27.4%
game-sm-success_2-5	       1.209 ms	     966.1 μs     -20.1%
game-sm-success_2-6	       531.9 μs	     388.9 μs     -26.9%
multisig-sm-1		       824.3 μs	     627.8 μs     -23.8%
multisig-sm-2		       814.1 μs	     613.4 μs     -24.7%
multisig-sm-3		       821.4 μs	     621.0 μs     -24.4%
multisig-sm-4		       829.1 μs	     629.3 μs     -24.1%
multisig-sm-5		       1.070 ms	     854.3 μs     -20.2%
multisig-sm-6		       821.9 μs	     628.2 μs     -23.6%
multisig-sm-7		       811.2 μs	     615.8 μs     -24.1%
multisig-sm-8		       821.7 μs	     622.4 μs     -24.3%
multisig-sm-9		       829.0 μs	     631.7 μs     -23.8%
multisig-sm-10		       1.071 ms	     858.1 μs     -19.9%
ping-pong-1		       682.3 μs	     520.8 μs     -23.7%
ping-pong-2		       682.1 μs	     519.9 μs     -23.8%
ping-pong_2-1		       437.9 μs	     325.0 μs     -25.8%
prism-1			       379.8 μs	     277.8 μs     -26.9%
prism-2			       884.0 μs	     678.8 μs     -23.2%
prism-3			       764.6 μs	     582.2 μs     -23.9%
pubkey-1		       326.1 μs	     236.9 μs     -27.4%
stablecoin_1-1		       1.692 ms	     1.359 ms     -19.7%
stablecoin_1-2		       447.7 μs	     324.1 μs     -27.6%
stablecoin_1-3		       1.940 ms	     1.559 ms     -19.6%
stablecoin_1-4		       473.3 μs	     345.0 μs     -27.1%
stablecoin_1-5		       2.462 ms	     1.995 ms     -19.0%
stablecoin_1-6		       587.7 μs	     426.7 μs     -27.4%
stablecoin_2-1		       1.683 ms	     1.359 ms     -19.3%
stablecoin_2-2		       446.4 μs	     324.1 μs     -27.4%
stablecoin_2-3		       1.934 ms	     1.562 ms     -19.2%
stablecoin_2-4		       473.4 μs	     345.4 μs     -27.0%
token-account-1		       387.8 μs	     297.7 μs     -23.2%
token-account-2		       681.0 μs	     519.7 μs     -23.7%
uniswap-1		       766.3 μs	     607.9 μs     -20.7%
uniswap-2		       473.3 μs	     356.4 μs     -24.7%
uniswap-3		       3.147 ms	     2.538 ms     -19.4%
uniswap-4		       770.6 μs	     568.6 μs     -26.2%
uniswap-5		       2.249 ms	     1.792 ms     -20.3%
uniswap-6		       734.7 μs	     541.0 μs     -26.4%
vesting-1		       693.9 μs	     537.2 μs     -22.6%
```

File: plutus-core/index-envs/test/TestRAList.hs [hunk 1]
```diff
@@ -12,6 +12,9 @@ import Test.Tasty.QuickCheck
 instance Arbitrary a => Arbitrary (B.RAList a) where
     arbitrary = fromList <$> arbitrary
 
+-- FIXME: workaround because the tests were originally written with 0-index in mind
+unsafeIndexOnePlus e i = B.unsafeIndexOne e (i+1)
```
Comment (michaelpj) [inline plutus-core/index-envs/test/TestRAList.hs:16]: fixme!

**Pull request merged**